src/test/java/com/answerdigital/benhession/academy/answerkingweek2/controllers/CategoryControllerTest.java

```diff
@@ -0,0 +1,113 @@
+package com.answerdigital.benhession.academy.answerkingweek2.controllers;
+
+import com.answerdigital.benhession.academy.answerkingweek2.request.AddCategoryRequest;
+import com.answerdigital.benhession.academy.answerkingweek2.request.UpdateCategoryRequest;
+import com.answerdigital.benhession.academy.answerkingweek2.services.CategoryService;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.springframework.test.web.servlet.ResultActions;
+import org.springframework.test.web.servlet.request.MockHttpServletRequestBuilder;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+
+@ContextConfiguration(classes = {CategoryController.class})
+@ExtendWith(SpringExtension.class)
+class CategoryControllerTest {
+    @Autowired
+    private CategoryController categoryController;
+
+    @MockBean
+    private CategoryService categoryService;
+
+    /**
+     * Method under test: {@link CategoryController#addCategory(AddCategoryRequest)}
+     */
+    @Test
+    void testAddCategory() throws Exception {
+        MockHttpServletRequestBuilder contentTypeResult = MockMvcRequestBuilders.get("/")
+                .contentType(MediaType.APPLICATION_JSON);
+
+        ObjectMapper objectMapper = new ObjectMapper();
+        MockHttpServletRequestBuilder requestBuilder = contentTypeResult.content(
+                objectMapper.writeValueAsString(new AddCategoryRequest("Name", "The characteristics of someone or something")));
+        ResultActions actualPerformResult = MockMvcBuilders.standaloneSetup(categoryController)
+                .build()
+                .perform(requestBuilder);
+        actualPerformResult.andExpect(MockMvcResultMatchers.status().isNotFound());
+    }
+
+    /**
+     * Method under test: {@link CategoryController#addItemToCategory(Long, Long)}
+     */
+    @Test
+    void testAddItemToCategory() throws Exception {
+        MockHttpServletRequestBuilder requestBuilder = MockMvcRequestBuilders.put("/{categoryId}/additem/{itemId}", 123L,
```

Comment:
and I think this code can be simplified to this: 
    @Test
    @WithMockUser("paul")
    void testAddItemToCategory() throws Exception {    
        mockMvc.perform(put("/{categoryId}/additem/{itemId}", 123L, 123L))
               .andExpect(MockMvcResultMatchers.status().isNotFound());
    }